lib/Mojolicious.pm

```diff
@@ -0,0 +1,1217 @@
+
+sub _exception {
+  my ($next, $c) = @_;
+  local $SIG{__DIE__} = \&_die;
+  $c->helpers->reply->exception($@) unless eval { $next->(); 1 };
+}
+
+1;
+
+=encoding utf8
+
+=head1 NAME
+
+Mojolicious - Real-time web framework
+
+=head1 SYNOPSIS
+
+  # Application
+  package MyApp;
+  use Mojo::Base 'Mojolicious', -signatures;
+
+  # Route
+  sub startup ($self) {
+    $self->routes->get('/hello')->to('foo#hello');
+  }
+
+  # Controller
+  package MyApp::Controller::Foo;
+  use Mojo::Base 'Mojolicious::Controller', -signatures;
+
+  # Action
+  sub hello ($self) {
+    $self->render(text => 'Hello World!');
+  }
+
+=head1 DESCRIPTION
+
+An amazing real-time web framework built on top of the powerful L<Mojo> web development toolkit. With support for
+RESTful routes, plugins, commands, Perl-ish templates, content negotiation, session management, form validation,
+testing framework, static file server, C<CGI>/C<PSGI> detection, first class Unicode support and much more for you to
+discover.
+
+Take a look at our excellent documentation in L<Mojolicious::Guides>!
+
+=head1 HOOKS
+
+L<Mojolicious> will emit the following hooks in the listed order.
+
+=head2 before_command
+
+Emitted right before the application runs a command through the command line interface.
+
+  $app->hook(before_command => sub ($command, $args) {...});
+
+Useful for reconfiguring the application before running a command or to modify the behavior of a command. (Passed the
+command object and the command arguments)
+
+=head2 before_server_start
+
+Emitted right before the application server is started, for web servers that support it, which includes all the
+built-in ones.
+
+  $app->hook(before_server_start => sub ($server, $app) {...});
+
+Useful for reconfiguring application servers dynamically or collecting server diagnostics information. (Passed the
+server and application objects)
+
+=head2 after_build_tx
+
+Emitted right after the transaction is built and before the HTTP request gets parsed.
+
+  $app->hook(after_build_tx => sub ($tx, $app) {...});
+
+This is a very powerful hook and should not be used lightly, it makes some rather advanced features such as upload
+progress bars possible. Note that this hook will not work for embedded applications, because only the host application
+gets to build transactions. (Passed the transaction and application objects)
+
+=head2 around_dispatch
+
+Emitted right after a new request has been received and wraps around the whole dispatch process, so you have to
+manually forward to the next hook if you want to continue the chain. Default exception handling with
+L<Mojolicious::Plugin::DefaultHelpers/"reply-E<gt>exception"> is the first hook in the chain and a call to
+L</"dispatch"> the last, yours will be in between.
+
+  $app->hook(around_dispatch => sub ($next, $c) {
+    ...
+    $next->();
+    ...
+  });
+
+This is a very powerful hook and should not be used lightly, it allows you to, for example, customize application-wide
```

Comment:
Perhaps in the cookbook document an example of customizing application-wide exception handling.